### PR TITLE
Fixes to allow typesets in unsupported dtypes and fixed `einops_reduce`

### PR DESCRIPTION
**Pull request by vedpatwardhan**
1. Moved the typesets logic from the `_dtype_device_wrapper_creator` to `_get_dtypes` to ensure that this can be used for functions like `einops_reduce`, which use the `unsupported_dtypes` attribute. 
2. Replaced the `inputs_to_ivy_arrays` decorator in `einops_reduce` by `inputs_to_native_arrays` as `einops.reduce` accepts native arrays.
3. Specified the `unsupported_dtypes` for tensorflow and paddle for `einops_reduce` to get the tests passing

### CODE REVIEW

File: ivy/functional/ivy/data_type.py [hunk 1]
```diff
@@ -203,13 +203,26 @@ def _get_dtypes(fn, complement=True):
...
+    }
+
     for key, merge_fn, base in basic:
         if hasattr(fn, key):
             v = getattr(fn, key)
             # only einops allowed to be a dictionary
             if isinstance(v, dict):
                 v = v.get(ivy.current_backend_str(), base)
-
+            for i, d in enumerate(v):
```
Comment (RickSanchezStoic) [inline ivy/functional/ivy/data_type.py:223]: This is causing a minor bug, where the existing dictionary seems to be getting overwritten. Try replicating the exact operation that was being performed back in the func_wrapper and maybe that fixes it.

File: ivy/func_wrapper.py [hunk 2]
```diff
@@ -1260,36 +1260,6 @@ def _dtype_device_wrapper_creator(attrib, t):
...
-        typesets = {
-            "valid": ivy.valid_dtypes,
-            "numeric": ivy.valid_numeric_dtypes,
-            "float": ivy.valid_float_dtypes,
-            "integer": ivy.valid_int_dtypes,
-            "unsigned": ivy.valid_uint_dtypes,
-            "complex": ivy.valid_complex_dtypes,
-        }
-
-        for key, value in version_dict.items():
```
Comment (RickSanchezStoic) [inline ivy/func_wrapper.py:1272]: Try doing the exact operation back in the data_types and it should fix the bug.

**Review by RickSanchezStoic — CHANGES_REQUESTED** [review]
So if you go to jax's choleksy function, and fetch its unsupported_dtypes it will only fetch the complex ones, while ignoring the floats, I believe this is happening because the dictionary is getting overwritten. Try replicating the exact dictionary operation happening back in func_wrappers and it might fix it, though I still have a little doubt about moving the operation entirely there, let's hope this alone fixes it.

File: ivy/functional/ivy/data_type.py (hunk 1 above)
Comment (vedpatwardhan) [reply ivy/functional/ivy/data_type.py]: the bug was basically with the way we were doing the `x[:i] + y + x[i+1:]`, not sure why the same line was working on the `func_wrapper`, but anyway just pushed a fix for that and should be resolved now

File: ivy/func_wrapper.py (hunk 2 above)
Comment (vedpatwardhan) [reply ivy/func_wrapper.py:1272]: done

**Comment by vedpatwardhan** [comment]
> So if you go to jax's choleksy function, and fetch its unsupported_dtypes it will only fetch the complex ones, while ignoring the floats, I believe this is happening because the dictionary is getting overwritten. Try replicating the exact dictionary operation happening back in func_wrappers and it might fix it, though I still have a little doubt about moving the operation entirely there, let's hope this alone fixes it.

Thanks for reviewing @RickSanchezStoic, just pushed a fix for this and `cholesky` should be working now. Actually we'd need to move the operation to this function if we want to allow typesets for functions like `einops_reduce`. Please take a look and let me know if there's anything else to be done, thanks 😄

**Comment by vedpatwardhan** [comment]
just found another bug, would get that fixed now

**Comment by vedpatwardhan** [comment]
done

**Comment by RickSanchezStoic** [comment]
Haven't done deep looking, but looks good to me otherwise, will see later if any issues arise. You can merge this, thanks!

**Review by RickSanchezStoic — APPROVED** [review]

**Comment by vedpatwardhan** [comment]
awesome, thanks @RickSanchezStoic 😄

**Pull request merged**